### ISSUE TITLE
Spec for PMUI Tab-Switch Refactoring

### ISSUE THREAD

**Issue by donnie-msft** [NuGet/Home#10154]
@donnie-msft commented on [Wed Oct 14 2020](https://github.com/NuGet/Client.Engineering/pull/588)

---

@donnie-msft commented on [Thu Oct 15 2020](https://github.com/NuGet/Client.Engineering/pull/588#issuecomment-709643096)

> Thanks for the great write-up @donnie-msft 
> 
> Something that I was not able to find (or missed), when you open the PM UI and it ends up on the default tab, does the browse tab start loading? I assume the answer is no. 
> 
> How about the reverse? On the browse, does the installed tab equivalent data start loading? 
> I know that we do look at the existing installed packages in order to populate the number in the updates tab label, and I think this is one reason why initial PM UI loading feels heavy to users with machines that are not really powerful.

This spec has no effect on the existing behavior of automatically loading whichever tab is open when the window is first opened.

We always load the tab header counts, and perhaps could be more conservative with that. Do you feel it fits with this work or a future spec?

---

@nkolev92 commented on [Thu Oct 15 2020](https://github.com/NuGet/Client.Engineering/pull/588#issuecomment-709646231)

> This spec has no effect on the existing behavior of automatically loading whichever tab is open when the window is first opened.

Oh I didn't mean to suggest that. We read a setting currently and we default to that, I think that's great.

> We always load the tab header counts, and perhaps could be more conservative with that. Do you feel it fits with this work or a future spec?

I think it's ok for that to be a part of a future work item. 
Doesn't the tab header count update require fetching of all packages for that project/solution & checking whether they have any updates?

---

@donnie-msft commented on [Thu Oct 15 2020](https://github.com/NuGet/Client.Engineering/pull/588#issuecomment-709707000)

> Doesn't the tab header count update require fetching of all packages for that project/solution & checking whether they have any updates?

Yes, and Deprecation info!
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L947-L950

---

@nkolev92 commented on [Fri Oct 16 2020](https://github.com/NuGet/Client.Engineering/pull/588#issuecomment-710624742)

> Yes, and Deprecation info!
> https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs#L947-L950

Yuck :D 
So say you open the browse tab...wait for 20s, then open the installed tab, does it now get loaded quicker than just loading the installed tab directly? 
Basically, are we reusing the work that we did for figuring out the label count?
